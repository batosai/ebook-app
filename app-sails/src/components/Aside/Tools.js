import React, { Component, PropTypes as T } from 'react';
import IconButton from 'material-ui/IconButton';
import {
  Toolbar,
  ToolbarGroup,
  ToolbarTitle,
  ToolbarSeparator,
} from 'material-ui/Toolbar';
import NavigationExpandMoreIcon from 'material-ui/svg-icons/navigation/expand-more';
import AddIcon from 'material-ui/svg-icons/content/add';
import MenuItem from 'material-ui/MenuItem';
import IconMenu from 'material-ui/IconMenu';
import { fullWhite } from 'material-ui/styles/colors';

class Tools extends Component {
  state = {
    title: 'Library',
  };

  collectionToggle = id => {
    this.props.findCollectionByLibrary(id);
  };

  touchTap = library => {
    if (library !== undefined) {
      this.setState({
        title: library.name,
      });
      this.collectionToggle(library.id);
    } else {
      this.setState({
        title: 'Library',
      });
      this.collectionToggle();
    }
  };

  render = () => {
    return (
      <Toolbar
        style={{
          backgroundColor: 'transparent',
          marginTop: '-4px',
          padding: 0,
        }}
      >
        <ToolbarGroup>
          <ToolbarTitle text={this.state.title} style={{ color: fullWhite }} />
          <IconMenu
            anchorOrigin={{ horizontal: 'right', vertical: 'bottom' }}
            targetOrigin={{ horizontal: 'right', vertical: 'bottom' }}
            iconButtonElement={
              <IconButton>
                <NavigationExpandMoreIcon color={fullWhite} />
              </IconButton>
            }
          >
            <MenuItem
              primaryText="Library"
              onTouchTap={() => this.touchTap()}
            />
            {this.props.libraries.map(library => (
              <MenuItem
                key={library.id}
                primaryText={library.name}
                onTouchTap={() => this.touchTap(library)}
              />
            ))}
          </IconMenu>
          <ToolbarSeparator style={{ margin: 0 }} />
          <IconButton>
            <AddIcon color={fullWhite} />
          </IconButton>
        </ToolbarGroup>
      </Toolbar>
    );
  };
}

Tools.propTypes = {
  libraries: T.array,
  findCollectionByLibrary: T.func.isRequired,
};

Tools.defaultProps = {
  libraries: [],
};

export default Tools;
