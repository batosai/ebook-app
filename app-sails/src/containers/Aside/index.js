import { connect } from 'react-redux';
import Aside from '../../components/Aside';

import { aside, library, collection } from '../../actions';

const mapStateToProps = appState => {
  return {
    open: appState.aside.open,
    libraryId: appState.aside.library,
    collections: appState.collections,
  };
};

const mapDispatchToProps = {
  toggleAside: aside.dispatch,
  findLibraries: library.fetchAll.dispatch,
  findCollections: collection.fetchAll.dispatch,
};

export default connect(mapStateToProps, mapDispatchToProps)(Aside);
