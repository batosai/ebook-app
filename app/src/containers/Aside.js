import React, { Component, PropTypes as T } from 'react';
import { connect } from 'react-redux';
import AppBar from 'material-ui/AppBar';
import Drawer from 'material-ui/Drawer';
import IconButton from 'material-ui/IconButton';
import { Toolbar, ToolbarGroup, ToolbarTitle, ToolbarSeparator } from 'material-ui/Toolbar';
import NavigationClose from 'material-ui/svg-icons/navigation/close';
import NavigationExpandMoreIcon from 'material-ui/svg-icons/navigation/expand-more';
import AddIcon from 'material-ui/svg-icons/content/add';
import MenuItem from 'material-ui/MenuItem';
import IconMenu from 'material-ui/IconMenu';
import Library from '../components/Library';
import { fullWhite } from 'material-ui/styles/colors';

import { asideToggle } from '../actions/aside';
import { getLibraries } from '../actions/libraries';

const style = {
  root: {
    display: 'flex',
    flexWrap: 'wrap',
    justifyContent: 'space-around',
  },
  gridList: {
    width: '100%',
    height: '100%',
    overflowY: 'auto',
  },
};

const tilesData = [
  {
    id: 1,
    img: 'http://www.material-ui.com/images/grid-list/00-52-29-429_640.jpg',
    title: 'Breakfast',
    author: 'jill111',
  },
  {
    id: 2,
    img: 'http://www.material-ui.com/images/grid-list/burger-827309_640.jpg',
    title: 'Tasty burger',
    author: 'pashminu',
  },
  {
    id: 3,
    img: 'http://www.material-ui.com/images/grid-list/camera-813814_640.jpg',
    title: 'Camera',
    author: 'Danson67',
  },
  {
    id: 4,
    img: 'http://www.material-ui.com/images/grid-list/morning-819362_640.jpg',
    title: 'Morning',
    author: 'fancycrave1',
  },
  {
    id: 5,
    img: 'http://www.material-ui.com/images/grid-list/hats-829509_640.jpg',
    title: 'Hats',
    author: 'Hans',
  },
  {
    id: 6,
    img: 'http://www.material-ui.com/images/grid-list/honey-823614_640.jpg',
    title: 'Honey',
    author: 'fancycravel',
  },
  {
    id: 7,
    img: 'http://www.material-ui.com/images/grid-list/vegetables-790022_640.jpg',
    title: 'Vegetables',
    author: 'jill111',
  },
  {
    id: 8,
    img: 'http://www.material-ui.com/images/grid-list/water-plant-821293_640.jpg',
    title: 'Water plant',
    author: 'BkrmadtyaKarki',
  },
];

const Tools= (props) => (
  <Toolbar style={{backgroundColor: 'transparent', marginTop: '-4px', padding: 0}}>
    <ToolbarGroup>
      <ToolbarTitle text="Library" style={{color:fullWhite}} />
      <IconMenu
        anchorOrigin={{horizontal: 'right', vertical: 'bottom'}}
        targetOrigin={{horizontal: 'right', vertical: 'bottom'}}
        iconButtonElement={<IconButton><NavigationExpandMoreIcon color={fullWhite} /></IconButton>} >
        {props.libraries.map(library => (
            <MenuItem key={library.id} primaryText={library.name} />
        ))}
      </IconMenu>
      <ToolbarSeparator style={{margin: 0}} />
      <IconButton><AddIcon color={fullWhite} /></IconButton>
    </ToolbarGroup>
  </Toolbar>
);

class Aside extends Component {
  handleToggle = () => this.props.asideToggle();

  componentWillMount() {
    this.props.getLibraries();
  }

  render() {
    return (
      <Drawer open={this.props.open}>
        <AppBar
          onLeftIconButtonTouchTap={this.handleToggle}
          showMenuIconButton={true}
          iconElementRight={<Tools libraries={this.props.libraries} />}
          iconElementLeft={<IconButton><NavigationClose /></IconButton>} />

        <Library
          style={style}
          tiles={tilesData} />
      </Drawer>
    );
  }
}

Aside.propTypes = {
  open: T.bool,
  asideToggle: T.func.isRequired,
  libraries: T.array,
  getLibraries: T.func.isRequired,
};

Aside.defaultProps = {
  libraries: []
};

function mapStateToProps(appState) {
  return {
    open: appState.aside.open,
    libraries: appState.libraries,
  };
}

export default connect(mapStateToProps, {asideToggle, getLibraries})(Aside);
