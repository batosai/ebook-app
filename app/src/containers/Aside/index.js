import { connect } from 'react-redux';
import Aside from '../../components/Aside';

import { asideToggle, findLibraries, findCollections } from '../../actions';

const mapStateToProps = (appState) => {
  return {
    open: appState.aside.open,
    collections: appState.collections.items,
  };
};

const mapDispatchToProps = {
  asideToggle,
  findLibraries,
  findCollections
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Aside);
