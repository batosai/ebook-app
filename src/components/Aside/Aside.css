
nav a:hover {
  background-color: #181818;
  border-radius: 5px;
}
