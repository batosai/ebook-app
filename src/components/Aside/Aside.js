import React, { Component } from 'react';
import Footer from './Footer';
import { Link } from 'react-router';
import { connect } from 'react-redux';
import { updateBook } from '../../actions/items';
import { toogleUpdate } from '../../actions/list';
import { getCategories } from '../../actions/collections';
import './Aside.css';

import styles from './Aside.style';

class Aside extends Component {

  onDropTarget(link, e) {
    let item = JSON.parse(e.dataTransfer.getData("item"));

    item.collection = link.slug;
    this.props.updateBook(item, () => {
      this.props.toogleUpdate();
    });
    e.preventDefault();
  }

  renderLinks() {
    return this.props.categories.map(link => (
      <li style={ styles.list.item } onDragOver={ (e) => e.preventDefault() } onDrop={ e => this.onDropTarget(link, e) } key={ link.id }>
        <Link to={ link.href } style={ styles.list.item.link } activeStyle={ styles.list.item.link.hover }>{ link.name }</Link>
      </li>
    ));
  }

  render() {
    return (
      <nav style={ styles.content }>
        <ul style={ styles.list }>
          { this.renderLinks() }
        </ul>
        <Footer />
      </nav>
    );
  }

  componentWillMount() {
    this.props.getCategories();
  }
}

Aside.propTypes = {
  getCategories: React.PropTypes.func.isRequired,
  categories : React.PropTypes.arrayOf(React.PropTypes.object).isRequired,
};

function mapStateToProps(appState) {
  return {
    categories: appState.collections,
    ...appState,
  };
}

export default connect(mapStateToProps, { updateBook, toogleUpdate, getCategories })(Aside);
