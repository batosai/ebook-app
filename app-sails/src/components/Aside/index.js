import React, { Component, PropTypes as T } from 'react';
import AppBar from 'material-ui/AppBar';
import Drawer from 'material-ui/Drawer';
import IconButton from 'material-ui/IconButton';
import NavigationClose from 'material-ui/svg-icons/navigation/close';
import Library from '../Library';
import Tools from '../../containers/Aside/Tools';

const style = {
  root: {
    display: 'flex',
    flexWrap: 'wrap',
    justifyContent: 'space-around',
  },
  gridList: {
    width: '100%',
    height: '100%',
    overflowY: 'auto',
  },
};

class Aside extends Component {
  handleToggle = () => this.props.toggleAside();

  componentWillMount = () => {
    // TODO lancer Ã  l'init, pendant le splashscreen
    this.props.findLibraries();
    this.props.findCollections();
  };

  render = () => {
    return (
      <Drawer open={this.props.open}>
        <AppBar
          onLeftIconButtonTouchTap={this.handleToggle}
          showMenuIconButton={true}
          iconElementRight={<Tools libraries={this.props.libraries} />}
          iconElementLeft={
            <IconButton>
              <NavigationClose />
            </IconButton>
          }
        />

        <Library style={style} tiles={this.props.collections} />
      </Drawer>
    );
  };
}

Aside.propTypes = {
  open: T.bool,
  toggleAside: T.func.isRequired,
  collections: T.array,
  findLibraries: T.func.isRequired,
  findCollections: T.func.isRequired,
};

Aside.defaultProps = {
  collections: [],
};

export default Aside;
