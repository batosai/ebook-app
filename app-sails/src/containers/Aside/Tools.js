import { connect } from 'react-redux';
import Tools from '../../components/Aside/Tools';

import { collection } from '../../actions';

const mapStateToProps = appState => {
  return {
    libraries: appState.libraries,
  };
};

const mapDispatchToProps = {
  findCollectionByLibrary: libraryId =>
    collection.fetchAll.dispatch({ library: libraryId }),
};

export default connect(mapStateToProps, mapDispatchToProps)(Tools);
