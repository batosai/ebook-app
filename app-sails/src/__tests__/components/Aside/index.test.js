import React from 'react';
import Aside from '../../../components/Aside';
import { shallow, mount, render } from 'enzyme';

function setup() {
  const props = {
    open: true,
    toggleAside: jest.fn(),
    collections: [],
    findLibraries: jest.fn(),
    findCollections: jest.fn(),
  };

  const enzymeWrapper = shallow(<Aside {...props} />);

  return {
    props,
    enzymeWrapper,
  };
}

describe('Components', () => {
  it('Aside', () => {
    const { enzymeWrapper } = setup();

    expect(enzymeWrapper).toMatchSnapshot();
  });
});
