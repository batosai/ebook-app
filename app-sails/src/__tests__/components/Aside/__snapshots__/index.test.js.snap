exports[`Components Aside 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <Drawer
    disableSwipeToOpen={false}
    docked={true}
    open={true}
    openSecondary={false}
    swipeAreaWidth={30}
    width={null}
    zDepth={2}>
    <AppBar
        iconElementLeft={
            <IconButton
                disableTouchRipple={false}
                disabled={false}
                iconStyle={Object {}}
                tooltipPosition="bottom-center"
                touch={false}>
                <NavigationClose />
            </IconButton>
        }
        iconElementRight={
            <Connect(Tools)
                libraries={undefined} />
        }
        onLeftIconButtonTouchTap={[Function]}
        showMenuIconButton={true}
        title=""
        zDepth={1} />
    <Library
        style={
            Object {
                "gridList": Object {
                  "height": "100%",
                  "overflowY": "auto",
                  "width": "100%",
                },
                "root": Object {
                  "display": "flex",
                  "flexWrap": "wrap",
                  "justifyContent": "space-around",
                },
              }
        }
        tiles={Array []} />
</Drawer>,
  "nodes": Array [
    <Drawer
      disableSwipeToOpen={false}
      docked={true}
      open={true}
      openSecondary={false}
      swipeAreaWidth={30}
      width={null}
      zDepth={2}>
      <AppBar
            iconElementLeft={
                  <IconButton
                        disableTouchRipple={false}
                        disabled={false}
                        iconStyle={Object {}}
                        tooltipPosition="bottom-center"
                        touch={false}>
                        <NavigationClose />
                  </IconButton>
            }
            iconElementRight={
                  <Connect(Tools)
                        libraries={undefined} />
            }
            onLeftIconButtonTouchTap={[Function]}
            showMenuIconButton={true}
            title=""
            zDepth={1} />
      <Library
            style={
                  Object {
                        "gridList": Object {
                          "height": "100%",
                          "overflowY": "auto",
                          "width": "100%",
                        },
                        "root": Object {
                          "display": "flex",
                          "flexWrap": "wrap",
                          "justifyContent": "space-around",
                        },
                      }
            }
            tiles={Array []} />
</Drawer>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <Aside
        collections={Array []}
        findCollections={[Function]}
        findLibraries={[Function]}
        open={true}
        toggleAside={[Function]} />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": Aside {
        "_reactInternalInstance": [Circular],
        "componentWillMount": [Function],
        "context": Object {},
        "handleToggle": [Function],
        "props": Object {
          "collections": Array [],
          "findCollections": [Function],
          "findLibraries": [Function],
          "open": true,
          "toggleAside": [Function],
        },
        "refs": Object {},
        "render": [Function],
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <Drawer
          disableSwipeToOpen={false}
          docked={true}
          open={true}
          openSecondary={false}
          swipeAreaWidth={30}
          width={null}
          zDepth={2}>
          <AppBar
                    iconElementLeft={
                              <IconButton
                                        disableTouchRipple={false}
                                        disabled={false}
                                        iconStyle={Object {}}
                                        tooltipPosition="bottom-center"
                                        touch={false}>
                                        <NavigationClose />
                              </IconButton>
                    }
                    iconElementRight={
                              <Connect(Tools)
                                        libraries={undefined} />
                    }
                    onLeftIconButtonTouchTap={[Function]}
                    showMenuIconButton={true}
                    title=""
                    zDepth={1} />
          <Library
                    style={
                              Object {
                                        "gridList": Object {
                                          "height": "100%",
                                          "overflowY": "auto",
                                          "width": "100%",
                                        },
                                        "root": Object {
                                          "display": "flex",
                                          "flexWrap": "wrap",
                                          "justifyContent": "space-around",
                                        },
                                      }
                    }
                    tiles={Array []} />
</Drawer>,
        "_debugID": 2,
        "_renderedOutput": <Drawer
          disableSwipeToOpen={false}
          docked={true}
          open={true}
          openSecondary={false}
          swipeAreaWidth={30}
          width={null}
          zDepth={2}>
          <AppBar
                    iconElementLeft={
                              <IconButton
                                        disableTouchRipple={false}
                                        disabled={false}
                                        iconStyle={Object {}}
                                        tooltipPosition="bottom-center"
                                        touch={false}>
                                        <NavigationClose />
                              </IconButton>
                    }
                    iconElementRight={
                              <Connect(Tools)
                                        libraries={undefined} />
                    }
                    onLeftIconButtonTouchTap={[Function]}
                    showMenuIconButton={true}
                    title=""
                    zDepth={1} />
          <Library
                    style={
                              Object {
                                        "gridList": Object {
                                          "height": "100%",
                                          "overflowY": "auto",
                                          "width": "100%",
                                        },
                                        "root": Object {
                                          "display": "flex",
                                          "flexWrap": "wrap",
                                          "justifyContent": "space-around",
                                        },
                                      }
                    }
                    tiles={Array []} />
</Drawer>,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Aside
    collections={Array []}
    findCollections={[Function]}
    findLibraries={[Function]}
    open={true}
    toggleAside={[Function]} />,
}
`;
